[imaging_browser] Now relying on getSetting->tblScanTypes for the New Data column

gluneau opened the pull request:
The new column is now in line with the tblScanTypes from the settings.

This removes hard coded AcquisitionProtocolID values from this module.

driusan commented on modules/imaging_browser/php/NDB_Menu_Filter_imaging_browser.class.inc:
```diff
@@ -124,7 +124,8 @@ class NDB_Menu_Filter_Imaging_Browser extends NDB_Menu_Filter
             WHERE OutputType='native' AND AcquisitionProtocolID 
-            NOT IN (1, 2, 3, 52) GROUP BY files.SessionID) nd 
+            IN (".
+            implode(",", array_keys($scan_id_types)) .") GROUP BY files.SessionID) nd
```
I don't know where $scan_id_types comes from, but this doesn't seem safe.

It looks like it's all scan id types that exist, making this in clause appear pointless.

Given that AcquisitionProtocolID 1, 2, and 3 don't exist because of the stupid default schema, and 52 in the default schema is 'scout' (which should probably be excluded at the protocol level, not in an imaging browser query), couldn't the same thing by achieved by just taking out the AcquisitionProtocolID part of the where clause?

gluneau replied:
$scan_id_types comes from line [81](https://github.com/gluneau/Loris/blob/774594cd9a8cef1d58d045bf0c1ffa7a400f96c3/modules/imaging_browser/php/NDB_Menu_Filter_imaging_browser.class.inc#L81) and got its values from the settings "tblScanTypes" at line [65](https://github.com/gluneau/Loris/blob/774594cd9a8cef1d58d045bf0c1ffa7a400f96c3/modules/imaging_browser/php/NDB_Menu_Filter_imaging_browser.class.inc#L65)

These are the protocols you use, then its the ones that will be re-used here.

johnsaigle commented:
Moving this to 17.2 as it was stated in the LORIS meeting that `Cleanup` PRs should not be going to 17.1 this close to the release.

gluneau commented:
@johnsaigle  This PR is important to the IBIS project since if it is merged in 17.1, this module will no longer need to be overridden.  Thank you for considering projects requirements as well.

johnsaigle commented:
Cool works for me.

xlecours commented:
This is effectively filtering out the 2 dti files from the raisinbread dataset if the only tblScanTypes are t1 and t2. If I add 'dti' in the tblScanTypes config list, the files are shown.

xlecours commented:
@driusan I am moving this into the `Senior dev review` lane because it is your call :)

MounaSafiHarab commented:
@xlecours @gluneau 

I removed the passed manual test because what @xlecours noted is indeed happening and it is not the intended behavior.

@gluneau 

1) if you select in the Imaging Module tab in Configuration a scan type type that you have no entries in the files table for (for example, pick mrs or scout), this will remove all the "rows" in imaging browser (so something wrong with the query as a result of the addition of f.AcquisitionProtocolID IN ... statement

(rows should not be impacted by what scan types you select; just columns)

2) if you select no mri_scan_type at all; then you will get an "error loading data" in imaging browser (and an error in SQL in the error_log)

gluneau commented:
@MounaSafiHarab @xlecours I've modified the code to allow for no selected scan type.

I've also removed the scan type from the where clause since we do not want to auto filter out stuff

xlecours commented:
@gluneau I am not sure what to test regarding the `New Data` column. The behaviour I see is an additional column in the table when a new tblScanTypes in the Configs.

MounaSafiHarab commented:
@gluneau 

Let me know when the New Data column filter is implemented so I can re-test

sruthymathew123 commented:
@leothomas 
PHP Notice:  Array to string conversion in /var/www/loris/php/libraries/Database.class.inc on line 688
PHP Notice:  Undefined index: ID in /var/www/loris/php/libraries/User.class.inc on line 72
PHP Notice:  Undefined index: Real_name in /var/www/loris/php/libraries/User.class.inc on line 86
PHP Notice:  Undefined variable: user_cid in /var/www/loris/php/libraries/User.class.inc on line 104
PHP Notice:  Undefined index: UserID in /var/www/loris/php/libraries/User.class.inc on line 217
PHP Fatal error:  Uncaught LorisException: Error: Database user named lorisadmin does not exist. Please create and then retry script
 in /var/www/loris/tools/fix_timepoint_date_problems.php:254
Stack trace:
#0 /var/www/loris/tools/fix_timepoint_date_problems.php(162): addInstrument('87', 'vineland_proban...')
#1 {main}
  thrown in /var/www/loris/tools/fix_timepoint_date_problems.php on line 254

MounaSafiHarab commented:
It is all good, except if you choose no scan types at all where you get an error loading data (it is an unlikely/impractical situation)

MounaSafiHarab commented:
tested it, and it works well now!
@driusan I added the new feature/add to release notes because projects somehow need to know that filtering for NEW now only returns session entries that have non-QC'ed imaging modalities coming from the Configuration module.
